browser/android/brave_shields_content_settings.cc

```diff
@@ -0,0 +1,134 @@
+
+#include <string>
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_string.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "components/content_settings/core/browser/content_settings_utils.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+
+#include "jni/BraveShieldsContentSettings_jni.h"
+
+static const char kAllowResource[] = "allow";
+
+namespace chrome {
+namespace android {
+
+BraveShieldsContentSettings* _brave_shields_content_settings = nullptr;
+
+static void JNI_BraveShieldsContentSettings_Init(JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller) {
+  _brave_shields_content_settings =
+    new BraveShieldsContentSettings(env, jcaller);
+}
+
+BraveShieldsContentSettings::BraveShieldsContentSettings(JNIEnv* env,
+    const base::android::JavaRef<jobject>& obj):
+    weak_java_native_worker_(env, obj) {
+  Java_BraveShieldsContentSettings_setNativePtr(env, obj,
+    reinterpret_cast<intptr_t>(this));
+}
+
+BraveShieldsContentSettings::~BraveShieldsContentSettings() {
+}
+
+void BraveShieldsContentSettings::Destroy(JNIEnv* env,
+        const base::android::JavaParamRef<jobject>& jcaller) {
+  _brave_shields_content_settings = nullptr;
+  delete this;
+}
+
+void BraveShieldsContentSettings::DispatchBlockedEventToJava(int tab_id,
+        const std::string& block_type, const std::string& subresource) {
+  JNIEnv* env = base::android::AttachCurrentThread();
+  Java_BraveShieldsContentSettings_blockedEvent(env,
+    weak_java_native_worker_.get(env), tab_id,
+    base::android::ConvertUTF8ToJavaString(env, block_type),
+    base::android::ConvertUTF8ToJavaString(env, subresource));
+}
+
+// static
+void BraveShieldsContentSettings::DispatchBlockedEvent(int tab_id,
+  const std::string& block_type, const std::string& subresource) {
+  DCHECK(_brave_shields_content_settings);
+  if (!_brave_shields_content_settings) {
+    return;
+  }
+  _brave_shields_content_settings->DispatchBlockedEventToJava(tab_id,
+    block_type, subresource);
+}
+
+base::android::ScopedJavaLocalRef<jstring>
+  JNI_BraveShieldsContentSettings_GetShields(
+    JNIEnv* env, jboolean incognito,
+    const base::android::JavaParamRef<jstring>& host,
+    const base::android::JavaParamRef<jstring>& resourceIndentifier) {
+  HostContentSettingsMap* map;
+  Profile* profile =
+    ProfileManager::GetActiveUserProfile()->GetOriginalProfile();
+  if (incognito) {
+    if (!profile->HasOffTheRecordProfile()) {
+      // Allow shields there
+      return base::android::ConvertUTF8ToJavaString(env, kAllowResource);
+    }
+    map = HostContentSettingsMapFactory::GetForProfile(
+        profile->GetOffTheRecordProfile());
+  } else {
+    map = HostContentSettingsMapFactory::GetForProfile(profile);
+  }
+
+  ContentSetting setting = map->GetContentSetting(
```

Comment:
I'm concerned that we're duplicating a lot of code from `DefaultBraveShieldsHandler`. I think we should consolidate this code in brave_shields_util and replace `GetShields`/`SetShields` with individual methods for adblock, etc...

Reply:
That part was refactored